src/os/windows/named_pipe/listener.rs

```diff
@@ -273,9 +285,24 @@ cannot create pipe server that has byte type but receives messages – have you
         let path = encode_to_utf16(self.path.as_os_str());
         let open_mode = self.open_mode(first, role, overlapped);
         let pipe_mode = self.pipe_mode(recv_mode, nonblocking);
-
         let mut sa = init_security_attributes();
         sa.bInheritHandle = 0;
+        if let Some(security_atributes) = &<Option<SecurityAttributes> as Clone>::clone(&self.security_attributes){
+            if let Some(_) = security_atributes.attributes {
+                match init_security_description() {
+                    Ok(security_descriptor) => {
+                        unsafe{
+                            (*(security_descriptor as *mut SECURITY_DESCRIPTOR)).Control = winapi::um::winnt::SE_DACL_PRESENT;
+                        }
+                        sa.lpSecurityDescriptor = security_descriptor;
+                    }
+                    Err(e) => {
+                        return Err(e);
+                    }
+                }
+
+            }
+        }
```

Comment:
This use of `winapi::um::winnt` leads me to believe that this code isn't entirely idiomatic in Windows terms. The official documentation mentions use of macros and the opaque nature of `SECURITY_DESCRIPTOR`, none of which is being respected here.